chapel-lang/chapel
# IFC: enable start_test-ing of the remaining U Col. tests

This removes the three remaining NOTEST files under
```
test/constrained-generics/ucol
```
enabling automatic testing for those directories.
I touched up the tests and added/edited/removed .futures as needed,
attempting to preserve the original intention of each test.

While there:

* Switch all testing of U Col. to compile with `--no-infer-implements-decls`
  This seems to have been a general assumption about how interfaces work
  at the time the tests were written.

* Remove `neelam/overloaded_min_with_implements.chpl`
  as a duplicate of `tier_3_prefer_constrained.chpl`

* Add `min_concept_void_promote.chpl`
  as a variant of `min_concept_auto_promote.chpl`
  The former should be legal whereas the latter should not.

In a couple of cases the test could be made to pass, however
I chose to make/leave it as a .future to highlight better behavior
or a design question.

Test changes, not reviewed.
